### DISCUSSION TITLE
Support for GEIST TOTEM in QMK

### DISCUSSION THREAD

**Discussion by bfkss** [manna-harbour/miryoku#209] [Q&A] [answered]
It would be great if you could add support for the [TOTEM](https://keeb.supply/products/geist-totem) to QMK.
There is already support for the TOTEM in [ZMK](https://github.com/manna-harbour/miryoku_zmk#38totem), but there are two variants of the TOTEM.

**Comment by manna-harbour** [comment, marked as answer]
Since totem isn't in QMK `master`, I won't add support to the `miryoku` branch yet. But I've added the necessary files to a temporary branch: https://github.com/manna-harbour/miryoku_qmk/tree/totem-test/keyboards/totem/keymaps/manna-harbour_miryoku. Also, since the totem keyboard definition provided by @GEIGEIGEIST (https://github.com/GEIGEIGEIST/qmk-config-totem) isn't in a QMK fork, I've added it in a temporary branch for ease of building: https://github.com/manna-harbour/miryoku_qmk/tree/totem-keyboard-test/keyboards/totem/.

Ordinarily you could build Miryoku QMK for totem by merging those 2 branches at build time using the Miryoku QMK build workflows, using `manna-harbour/miryoku_qmk/totem-keyboard-test manna-harbour/miryoku_qmk/totem-test` with the `merge` option (https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#build-inputs). However, at the moment, there is a merge conflict between the `miryoku` branch and QMK `master`, so the automatic merge will fail. I'll rebase the `miryoku` branch and comment here. Until then you could copy files or merge branches manually and build locally.

**Reply by manna-harbour** [reply]
The `miryoku` branch has been rebased, so building from the workflow as described above is now successful. Here's a sample build run: https://github.com/manna-harbour/miryoku_qmk/actions/runs/4900653848.
